src/Meziantou.Analyzer/Rules/UseShellExecuteAnalyzer.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Meziantou.Analyzer.Rules;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class UseShellExecuteAnalyzer : DiagnosticAnalyzer
+{
+    private static readonly DiagnosticDescriptor UseShellExecuteMustBeExplicitlySet = new(
+        RuleIdentifiers.UseShellExecuteMustBeSet,
+        title: "UseShellExecute must be explicitly set",
+        messageFormat: "UseShellExecute must be explicitly set when initializing a ProcessStartInfo",
+        RuleCategories.Usage,
+        DiagnosticSeverity.Warning,
+        isEnabledByDefault: true,
```

Comment:
I'm certain there will be numerous false positives because it doesn't recognize `processStartInfo.UseShellExecute = false;`. Implementing it flawlessly would likely necessitate a data flow analysis. Could you disable the rule by default? We can enable it by default later, once we gain more confidence.